### PR TITLE
enable logger using log4j2

### PR DESCRIPTION
**Pull request by ansyral**
i chose log4j rather than java.util.logging as the former is more popular and has more appenders(handler) like rollingfile

### CODE REVIEW

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/Logger.java [hunk 1]
```diff
@@ -23,16 +25,22 @@
...
+        inner.error(message, e);
     }
 
     public static void logError(String error) {
         JavaDebuggerServerPlugin.logError(error);
+        inner.error(error);
     }
     
+    public static void logWarn(String warn) {
+        inner.warn(warn);
```
Comment (andxu) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/Logger.java]: I think JavaDebuggerServerPlugin should log some thing(eg: logInfo if there is no warn method).

**Comment by andxu** [comment]
There are some codes with the same purpose, avoid to reinvent the wheel:
http://www.javatips.net/api/JCommon-master/logging/src/main/java/com/facebook/logging/DynamicLogger.java

**Comment by ansyral** [comment]
@andxu i don't think we need to reference the three-party library as it is just a design pattern, seems it doesn't provide more functionality?

**Comment by andxu** [comment]
I mean the api design/imp of logging in our code should take consideration of other implementations since    logging is very common, there are so many good examples for us.

**Review by andxu — DISMISSED** [review]

**Review by andxu — APPROVED** [review]

**Review by andxu — APPROVED** [review]

**Pull request closed without merging**